src/main/java/com/example/tradingapp/deribit/fix/DeribitFixClientAdapter.java

```diff
@@ -0,0 +1,86 @@
+package com.example.tradingapp.deribit.fix;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import quickfix.*;
+import quickfix.field.MsgType;
+
+@Slf4j
+@Component
+public class DeribitFixClientAdapter implements Application {
+
+    private final DeribitFixMessageCracker messageCracker;
+
+    public DeribitFixClientAdapter(DeribitFixMessageCracker messageCracker) {
+        this.messageCracker = messageCracker;
+    }
+
+    @Override
+    public void onCreate(SessionID sessionId) {
+        log.info("onCreate: SessionId={}", sessionId);
+    }
+
+    @Override
+    public void onLogon(SessionID sessionId) {
+        log.info("onLogon: SessionId={}", sessionId);
+    }
+
+    @Override
+    public void onLogout(SessionID sessionId) {
+        log.info("onLogout: SessionId={}", sessionId);
+    }
+
+    @Override
+    public void toAdmin(Message message, SessionID sessionID) {
+        log.info("to admin: message {}, sessionId {}", message, sessionID);
+        String msgType = null;
+        try {
+            msgType = message.getHeader().getString(MsgType.FIELD);
+        } catch (FieldNotFound e) {
+            throw new RuntimeException(e);
+        }
+        log.info("Message type to admin: {}", msgType);
+        try {
+            messageCracker.crack(message, sessionID);
+        } catch (UnsupportedMessageType | FieldNotFound | IncorrectTagValue e) {
+            log.error(e.getMessage());
+        }
```

Comment:
why did you split into 2 `catch` blocks and not handle `FieldNotFound` the same way in both variants?